### PR TITLE
RNN API cleanup for cuDNN 9.1

### PR DESCRIPTION
**Pull request by eqy**
Can potentially avoid a bit of boilerplate if we move directly to cuDNN 9.1's RNN API...

cc @csarofeen @ptrblck @xwang233 @zou3519 @mikaylagawarecki

### CODE REVIEW

**Comment by bdhirsh** [comment]
Marking triaged but feel free to ping someone when you want a review

File: aten/src/ATen/native/cudnn/RNN.cpp [hunk 1]
```diff
@@ -749,8 +745,8 @@ get_parameters(
                 linear_id,
                 lin_layer_mat_desc.mut_desc(),
                 &matrix_pointer,
-                unused_desc.mut_desc(),
-                &unused_pointer));
+                NULL,
```
Comment (Skylion007) [inline aten/src/ATen/native/cudnn/RNN.cpp:748]:
```suggestion
                nullptr,
```

**Review by Skylion007 — APPROVED** [review]

**Comment by eqy** [comment]
unlucky: https://github.com/pytorch/pytorch/issues/113908

**Pull request closed without merging**